### PR TITLE
Parameterize references to generic types. 

### PR DESCRIPTION
**Pull request by TruongNhanNguyen**
In this PR, I have solved the problem in Java implementation that related to _parameterize references to generic types_ by adding types to generic containers references.  

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

**Review by vil02 — CHANGES_REQUESTED** [review]
Please _revert_ all of the changes in the doc-strings - they are out of scope of this PR and we do not have a fixed style of them.

File: src/main/java/com/thealgorithms/datastructures/dynamicarray/DynamicArray.java [hunk 1]
```diff
@@ -120,6 +120,7 @@ private void fastRemove(final Object[] elements, final int index) {
         elements[this.size = newSize] = null;
     }
 
+    @SuppressWarnings("unchecked")
```
[the remark this answers is not in the document]
Comment (TruongNhanNguyen) [reply src/main/java/com/thealgorithms/datastructures/dynamicarray/DynamicArray.java]: I have removed the suppression as you requested.

File: src/main/java/com/thealgorithms/datastructures/lists/CircleLinkedList.java [hunk 2]
```diff
@@ -91,7 +94,7 @@ public E remove(int pos) {
     }
 
     public static void main(String[] args) {
-        CircleLinkedList cl = new CircleLinkedList<String>();
+        CircleLinkedList<Integer> cl = new CircleLinkedList<Integer>();
```
[the remark this answers is not in the document]
Comment (TruongNhanNguyen) [reply src/main/java/com/thealgorithms/datastructures/lists/CircleLinkedList.java]: I have remove boilerplate generic type `Integer` in the constructor.

File: src/main/java/com/thealgorithms/misc/ThreeSumProblem.java [hunk 3]
```diff
@@ -29,7 +29,7 @@ public List<List<Integer>> BruteForce(int[] nums, int target) {
             for (int j = i + 1; j < nums.length; j++) {
                 for (int k = j + 1; k < nums.length; k++) {
                     if (nums[i] + nums[j] + nums[k] == target) {
-                        List<Integer> temp = new ArrayList<>();
+                        List<Integer> temp = new ArrayList<Integer>();
```
[the remark this answers is not in the document]
Comment (TruongNhanNguyen) [reply src/main/java/com/thealgorithms/misc/ThreeSumProblem.java]: I have remove boilerplate generic type `Integer` in the constructor.

File: src/main/java/com/thealgorithms/misc/ThreeSumProblem.java [hunk 4]
```diff
@@ -54,7 +54,7 @@ public List<List<Integer>> TwoPointer(int[] nums, int target) {
             end = nums.length - 1;
             while (start < end) {
                 if (nums[start] + nums[end] + nums[i] == target) {
-                    List<Integer> temp = new ArrayList<>();
+                    List<Integer> temp = new ArrayList<Integer>();
```
[the remark this answers is not in the document]
Comment (TruongNhanNguyen) [reply src/main/java/com/thealgorithms/misc/ThreeSumProblem.java]: I have remove boilerplate generic type `Integer` in the constructor.

File: src/main/java/com/thealgorithms/misc/ThreeSumProblem.java [hunk 5]
```diff
@@ -75,8 +75,8 @@ public List<List<Integer>> TwoPointer(int[] nums, int target) {
 
     public List<List<Integer>> Hashmap(int[] nums, int target) {
         Arrays.sort(nums);
-        Set<List<Integer>> ts = new HashSet();
-        HashMap<Integer, Integer> hm = new HashMap<>();
+        Set<List<Integer>> ts = new HashSet<List<Integer>>();
+        HashMap<Integer, Integer> hm = new HashMap<Integer, Integer>();
```
[the remark this answers is not in the document]
Comment (TruongNhanNguyen) [reply src/main/java/com/thealgorithms/misc/ThreeSumProblem.java]: I have removed boilerplate generic types `Integer` and in the constructor.

File: src/main/java/com/thealgorithms/misc/ThreeSumProblem.java [hunk 6]
```diff
@@ -86,14 +86,14 @@ public List<List<Integer>> Hashmap(int[] nums, int target) {
             for (int j = i + 1; j < nums.length; j++) {
                 int t = target - nums[i] - nums[j];
                 if (hm.containsKey(t) && hm.get(t) > j) {
-                    List<Integer> temp = new ArrayList<>();
+                    List<Integer> temp = new ArrayList<Integer>();
```
[the remark this answers is not in the document]
Comment (TruongNhanNguyen) [reply src/main/java/com/thealgorithms/misc/ThreeSumProblem.java]: I have removed boilerplate generic type `Integer` in the constructor.

File: src/main/java/com/thealgorithms/misc/ThreeSumProblem.java [hunk 7]
```diff
@@ -86,14 +86,14 @@ public List<List<Integer>> Hashmap(int[] nums, int target) {
...
                     temp.add(nums[i]);
                     temp.add(nums[j]);
                     temp.add(t);
                     ts.add(temp);
                 }
             }
         }
-        return new ArrayList(ts);
+        return new ArrayList<List<Integer>>(ts);
```
[the remark this answers is not in the document]
Comment (TruongNhanNguyen) [reply src/main/java/com/thealgorithms/misc/ThreeSumProblem.java]: I have removed boilerplate generic type `List<Integer>` in the constructor.

File: src/main/java/com/thealgorithms/searches/UnionFind.java [hunk 8]
```diff
@@ -45,7 +45,7 @@ public void union(int x, int y) {
     }
 
     public int count() {
-        List parents = new ArrayList();
+        List<Integer> parents = new ArrayList<Integer>();
```
[the remark this answers is not in the document]
Comment (TruongNhanNguyen) [reply src/main/java/com/thealgorithms/searches/UnionFind.java]: I have removed boilerplate generic type `Integer` in the constructor.

File: src/main/java/com/thealgorithms/strings/WordLadder.java [hunk 9]
```diff
@@ -44,20 +44,20 @@ class WordLadder {
...
     public static int ladderLength(String beginWord, String endWord, List<String> wordList) {
-        HashSet<String> set = new HashSet(wordList);
+        HashSet<String> set = new HashSet<String>(wordList);
 
         if (!set.contains(endWord)) {
             return 0;
         }
 
-        Queue<String> queue = new LinkedList();
+        Queue<String> queue = new LinkedList<String>();
```
[the remark this answers is not in the document]
Comment (TruongNhanNguyen) [reply src/main/java/com/thealgorithms/strings/WordLadder.java]: I have removed boilerplate generic type `String` in the constructor.

**Comment by vil02** [comment]
> > Please _revert_ all of the changes in the doc-strings - they are out of scope of this PR and we do not have a fixed style of them.
> 
> Well, I see that the doc-strings were automatedly formatted by the Java Extension and it follow the best practices of a Java Docstring. Although, the doc-string changes are out of the scope of this PR, I think this is good and look better for code readability.

Then please deactivate this Java Extension. As I said before: this repository does not have any fixed and defined style of doc comments. Introducing a new style in few files, would only make the things more messy.

**Comment by TruongNhanNguyen** [comment]
> > > Please _revert_ all of the changes in the doc-strings - they are out of scope of this PR and we do not have a fixed style of them.
> > 
> > 
> > Well, I see that the doc-strings were automatedly formatted by the Java Extension and it follow the best practices of a Java Docstring. Although, the doc-string changes are out of the scope of this PR, I think this is good and look better for code readability.
> 
> Then please deactivate this Java Extension. As I said before: this repository does not have any fixed and defined style of doc comments. Introducing a new style in few files, would only make the things more messy.

Thanks for your clarification. I have reverted doc-strings as you requested.

**Review by vil02 — APPROVED** [review]

**Pull request merged**